Investigate Intermittent Failure in TestSigsExchange Test

## 🎯 Problem to be solved

The `TestSigsExchange` test in the `charon/dkg` package is failing intermittently in CI and on some developer machines. The failures seem to be related to a recent upgrade to `libp2p` `0.33`, which defaults to using `TLS`. The test behavior is inconsistent, sometimes completing quickly and sometimes never finishing.

TestSigsExchange
```
        	Error Trace:	/Users/kalo/obol/charon/dkg/nodesigs_internal_test.go:100
        	Error:      	Received unexpected error:
        	            	k1 lock hash signature broadcast: send sig request: context canceled
        	Test:       	TestSigsExchange
```
TestSyncFlow
```
--- PASS: TestSyncFlow/four_connect_two_disconnect (5.18s)
--- PASS: TestSyncFlow (10.27s)
PASS
ok      github.com/obolnetwork/charon/dkg       10.545s
```





```
 nodesigs_internal_test.go:100:
                Error Trace:    /home/gsora/Documents/Work/charon/dkg/nodesigs_internal_test.go:100
                Error:          Received unexpected error:
                                k1 lock hash signature broadcast: send sig request: new stream: failed to dial: failed to dial 16Uiu2HAmRd7HmCPpyz6H7
iE3Nrs4jrhwW7fgtFWWRwfKcyEAXPtU: all dials failed
                                  * [/ip4/127.0.0.1/tcp/39235] failed to negotiate security protocol: tls: received unexpected handshake message of t
ype *tls.clientHelloMsg when waiting for *tls.serverHelloMsg
                Test:           TestSigsExchange
--- FAIL: TestSigsExchange (15.00s)
FAIL
exit status 1
FAIL    github.com/obolnetwork/charon/dkg       15.009s

```
## 🛠️ Proposed solution

- [ ] Investigate the root cause of the intermittent test failures
- [ ]  Determine if the issue is directly related to the libp2p upgrade, and if so, explore potential workarounds or fixes

--- a/testutil/random.go
+++ b/testutil/random.go
@@ -30,6 +30,7 @@ import (
 	"github.com/libp2p/go-libp2p"
 	p2pcrypto "github.com/libp2p/go-libp2p/core/crypto"
 	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/p2p/transport/tcp"
 	"github.com/multiformats/go-multiaddr"
 	"github.com/prysmaticlabs/go-bitfield"
 	"github.com/stretchr/testify/require"
@@ -1180,6 +1181,7 @@ func CreateHostWithIdentity(t *testing.T, addr *net.TCPAddr, secret *k1.PrivateK
 	opts2 := []libp2p.Option{
 		libp2p.Identity((*p2pcrypto.Secp256k1PrivateKey)(secret)),
 		libp2p.ListenAddrs(addrs),
+		libp2p.Transport(tcp.NewTCPTransport, tcp.DisableReuseport()),
 	}
 	opts2 = append(opts2, opts...)
 

